docs/datatypes/matrices.md

```diff
@@ -224,6 +227,27 @@ If you have a matrix where the first dimension means `x` and the second
 means `y`, this will look confusing since `x` is printed as _column_ 
 (vertically) and `y` as _row_ (horizontally).
 
+## Not rectangular arrays
+
+By nesting arrays it is possible to have arrays that are not rectangular, for example.
+```js
+[[1, 2], [3, 4]] // rectangular of size [2, 2]
+[[1, 2], [3]]    // jagged
+[[1, 2], 3]      // heterogeneous
+```
+
+These types of arrays can't be converted to a matrix, but many operations are available for them.
+```js
+const A = [[1, 2], 3]
+math.add(A, 1)
+math.map(A, a => a+1)
+math.forEach(A, a => console.log(a))
+```
+Many matrix functions expect a rectangular array and might provide unexpected results with non rectangular arrays, for example.
+```js
+math.size([[1, 2], [3]]) // [2, 2]
+```
+The process of validation for rectangularity is expensive and this provides a fast way of working with nested arrays of many kinds.
```

Comment:
The antecedent of "this" in "this provides" is unclear. Please clarify exactly what provides a fast way of working with nested arrays.